src/libraries/System.Private.CoreLib/src/System/Buffers/Text/FormattingHelpers.CountDigits.cs

```diff
@@ -24,8 +24,9 @@ public static int CountDigits(ulong value)
             ];
             Debug.Assert(log2ToPow10.Length == 64);
 
+            // We use 'nint' because it gives us a free early zero-extension to 64 bits when running on a 64-bit platform.
```

Comment:
```suggestion
            // We use a native-sized integer because it gives us a free early zero-extension to 64 bits when running on a 64-bit platform.
```